### ISSUE TITLE
threadsafe_timeseries.partial_history unit test takes too long

### ISSUE THREAD

**Issue by vincentberenz** [machines-in-motion/real_time_tools#13]
@luator 

takes around 20 seconds on my machines. If possible, should be shorter. (I guess we can live with it otherwise). not sure who I should assign.

**Comment by luator** [comment]
I looked at the code of the test.  In the `partial_history` case, the `slow` flag is set to true which results in adding a 1ms sleep in a loop [here](https://github.com/machines-in-motion/real_time_tools/blob/b4e0831a822538070ea1f68e2f37af2f94d5d08c/tests/threadsafe_timeseries_test.cpp#L119).

There are two options to speed up the test:
1) reduce the sleep duration (might not be desired for this test?)
2) reduce the length of the input data (=number of iterations of the loop containing the sleep).  Currently the length is hard-coded [here](https://github.com/machines-in-motion/real_time_tools/blob/b4e0831a822538070ea1f68e2f37af2f94d5d08c/tests/threadsafe_timeseries_test.cpp#L25).

I am not familiar with this test. Would reducing the length be okay or would this invalidate the test in some way?  @mwuethri maybe you can take a look, this seems to be at least partially coming from you.

**Comment by wumanu** [comment]
yes, reducing the data length should be fine (as long as it is longer than the time-series).

i am not familiar with this test anymore because @MaximilienNaveau rewrote it almost completely. i think he also added the extended sleep, which makes it so slow, to prevent issues on non-realtime machines. @MaximilienNaveau could you have a look at the issue above?
